src/app/notebook/open/open.component.ts

```diff
@@ -23,9 +25,20 @@ export class OpenComponent {
   async importNotebook(event: Event) {
     const input = event.target as HTMLInputElement;
     if (input.files && input.files.length > 0) {
-      const blocks = JSON.parse(await input.files[0].text()).blocks;
-      await this.manager.bulkInsertBlocks(blocks);
+      const currentBlocks = (await this.manager.exportNotebook(url.read("p"), url.read("t"))).blocks;
+      let currentMaxndex = currentBlocks.reduce((max, _, index) => Math.max(max, index), 0);
```

Comment:
It is unnecessary because you can use:
```currentMaxIndex = currentBlocks.lenght-1; ```
Blocks have their own index because they are stored randomly in the database, but they are presented in order to the user. So, if you want to store new blocks after the current ones, you only need the number of blocks.

Reply:
It's hard exactly to remember what the rationale behind this was, but I believe somehow the numbers didn't match, and I wanted to use something different and was experimenting with getting the correct numbers for the place to insert. Seems I will have to change my approach either way.